src/main/java/org/reactome/addlinks/referencecreators/SimpleReferenceCreator.java

```diff
@@ -109,61 +109,67 @@ public void createIdentifiers(long personID, Map<String, T> mapping, List<GKInst
 
 		List<String> thingsToCreate = Collections.synchronizedList(new ArrayList<String>());
 
-		sourceReferences.parallelStream().forEach(sourceReference ->
+		if (mapping == null)
 		{
-			try
+			logger.error("Input mapping is null. Something must have gone wrong in the file processing stage.");
+		}
+		else
+		{
+			sourceReferences.parallelStream().forEach(sourceReference ->
 			{
-				String sourceReferenceIdentifier = (String) sourceReference.getAttributeValue(this.sourceIdentifyingAttribute);
-
-				if (mapping.containsKey(sourceReferenceIdentifier))
+				try
 				{
-					// It's possible that we could get a list of things from some third-party that contains mappings for multiple species.
-					// So we need to get the species for EACH thing we iterate on. I worry this will slow it down, but  it needs to be done
-					// if we want new identifiers to have the same species of the thing which they refer to.
-					Long speciesID = null;
-					@SuppressWarnings("unchecked")
-					Collection<GKSchemaAttribute> attributes = (Collection<GKSchemaAttribute>) sourceReference.getSchemClass().getAttributes();
-					if ( attributes.stream().filter(attr -> attr.getName().equals(ReactomeJavaConstants.species)).findFirst().isPresent())
+					String sourceReferenceIdentifier = (String) sourceReference.getAttributeValue(this.sourceIdentifyingAttribute);
+
+					if (mapping.containsKey(sourceReferenceIdentifier))
 					{
-						GKInstance speciesInst = (GKInstance) sourceReference.getAttributeValue(ReactomeJavaConstants.species);
-						if (speciesInst != null)
+						// It's possible that we could get a list of things from some third-party that contains mappings for multiple species.
+						// So we need to get the species for EACH thing we iterate on. I worry this will slow it down, but  it needs to be done
+						// if we want new identifiers to have the same species of the thing which they refer to.
+						Long speciesID = null;
+						@SuppressWarnings("unchecked")
+						Collection<GKSchemaAttribute> attributes = (Collection<GKSchemaAttribute>) sourceReference.getSchemClass().getAttributes();
+						if ( attributes.stream().filter(attr -> attr.getName().equals(ReactomeJavaConstants.species)).findFirst().isPresent())
```

Comment:
Here you could use `sourceReference.getSchemClass().getAttribute(ReactomeJavaConstants.species)` to get the species attribute directly rather than streaming and filtering.

Reply:
This is a white-space only change and not in scope for this PR.

Reply:
To see the non-whitespace changes of this PR: https://github.com/reactome/add-links/pull/165/files?diff=unified&w=1

Reply:
Okay, yeah - I think the PR looks good taking into account non-whitespace only.  If you want you could this in future, though.